main.go

```diff
@@ -110,6 +112,24 @@ func mainWithExitCode() exitCode {
 		return code
 	}
 
+	if args.enableGoRuntimeProfiler {
+		addr, _ := strings.CutPrefix(args.agentURL, "http://")
+		err = profiler.Start(
+			profiler.WithService(args.serviceName+"-profiler"),
+			profiler.WithEnv(args.environment),
+			profiler.WithVersion(versionInfo.Version),
+			profiler.WithAgentAddr(addr),
+			profiler.WithProfileTypes(
+				profiler.CPUProfile,
+				profiler.HeapProfile,
```

Comment:
I would also enable those profile types which have very low overhead:

```suggestion
				profiler.HeapProfile,
				profiler.GoroutineProfile,
				profiler.MutexProfile,
```